ext/openssl/ossl_pkey_rsa.c

```diff
@@ -628,6 +628,153 @@ ossl_rsa_to_public_key(VALUE self)
     return obj;
 }
 
+#if (OPENSSL_VERSION_NUMBER >= 0x10000000)
+/*
+ * call-seq:
+ *    rsa.sign_pss -> String
+ *
+ * Probabilistic Signature Scheme for RSA sign().
+ *
+ * To sign the +String+ +data+, +digest+, an instance of OpenSSL::Digest, must
+ * be provided. The return value is again a +String+ containing the signature.
+ * A PKeyError is raised should errors occur.
+ * Any previous state of the +Digest+ instance is irrelevant to the signature
+ * outcome, the digest instance is reset to its initial state during the
+ * operation.
+ *
+ * == Example
+ *   data = 'Sign me!'
+ *   digest = OpenSSL::Digest::SHA256.new
+ *   pkey = OpenSSL::PKey::RSA.new(2048)
+ *   signature = pkey.sign_pss(digest, data)
+ */
+static VALUE
+ossl_rsa_sign_pss(VALUE self, VALUE digest, VALUE data)
+{
+    EVP_PKEY *pkey;
+    EVP_PKEY_CTX *pkey_ctx;
+    const EVP_MD *md;
+    EVP_MD_CTX *md_ctx;
+    size_t buf_len;
+    VALUE signature;
+    int result;
+
+    pkey = GetPrivPKeyPtr(self);
+    md = GetDigestPtr(digest);
+    StringValue(data);
+    signature = rb_str_new(0, EVP_PKEY_size(pkey)+16);
```

Comment:
hmm. I lifted that directly from the original `sign()` method in `ext/openssl/ossl_pkey.c`

Are you saying that because this is a RSA-specific `pkey` its size is sufficient for the RSA signature? That would make sense to me.